File: chatbot.swift
// IOS Version
import OpenAI

let apiKey = "MY_API_KEY"
OpenAI.apiKey = apiKey

var messages = [[String: String]]()
let systemMsg = "Write Prompt"
messages.append(["role": "system", "content": systemMsg])

print("Your new assistant is ready!")
var userInput = ""
while userInput != "quit()" {
    if let input = readLine() {
        userInput = input
        messages.append(["role": "user", "content": input])
        let response = try? OpenAI.ChatCompletion.create(
            model: "gpt-3.5-turbo",
            messages: messages)
        if let reply = response?.choices[0].message.content {
            messages.append(["role": "assistant", "content": reply])
            print("\n\(reply)\n")
        }
    }
}

File: chatbot.cs
using System;
using System.Collections.Generic;
using OpenAI;

class Program
{
    static void Main(string[] args)
    {
        string apiKey = "MY_API_KEY";
        OpenAI.APIKey = apiKey;

        var messages = new List<Dictionary<string, string>>();
        string systemMsg = "Write Prompt";
        messages.Add(new Dictionary<string, string> { { "role", "system" }, { "content", systemMsg } });

        Console.WriteLine("Your new assistant is ready!");
        string userInput = "";
        while (userInput != "quit()")
        {
            userInput = Console.ReadLine();
            messages.Add(new Dictionary<string, string> { { "role", "user" }, { "content", userInput } });
            var response = OpenAI.ChatCompletion.Create(
                model: "gpt-3.5-turbo",
                messages: messages);
            string reply = response.Choices[0].Message.Content;
            messages.Add(new Dictionary<string, string> { { "role", "assistant" }, { "content", reply } });
            Console.WriteLine($"\n{reply}\n");
        }
    }
}
